src/test/java/uk/gov/hmcts/divorce/citizen/event/CitizenInviteApplicant2Test.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.divorce.citizen.event;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.ccd.sdk.ConfigBuilderImpl;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.common.model.CaseData;
+import uk.gov.hmcts.divorce.common.model.State;
+import uk.gov.hmcts.divorce.common.model.UserRole;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.divorce.citizen.event.CitizenInviteApplicant2.CITIZEN_INVITE_APPLICANT_2;
+import static uk.gov.hmcts.divorce.testutil.TestDataHelper.caseData;
+
+@ExtendWith(SpringExtension.class)
+public class CitizenInviteApplicant2Test {
+
+    @InjectMocks
+    private CitizenInviteApplicant2 citizenInviteApplicant2;
+
+    @Test
+    void shouldAddConfigurationToConfigBuilder() {
+        final Set<State> stateSet = Set.of(State.class.getEnumConstants());
+        final ConfigBuilderImpl<CaseData, State, UserRole> configBuilder = new ConfigBuilderImpl<>(CaseData.class, stateSet);
+
+        citizenInviteApplicant2.configure(configBuilder);
+
+        assertThat(configBuilder.getEvents().get(0).getId()).isEqualTo(CITIZEN_INVITE_APPLICANT_2);
+    }
+
+    @Test
+    public void givenValidCaseDataWhenCallbackIsInvokedThenGeneratePin() throws Exception {
+        final CaseData caseData = caseData();
+
+        final CaseDetails<CaseData, State> details = new CaseDetails<>();
+        details.setData(caseData);
+
+        final AboutToStartOrSubmitResponse<CaseData, State> response = citizenInviteApplicant2.aboutToSubmit(details, details);
+
+        assertThat(response.getData().getInvitePin()).isNotBlank();
+        assertThat(response.getData().getInvitePin().length()).isEqualTo(8);
+        assertThat(response.getData().getInvitePin().contains("I")).isFalse();
+        assertThat(response.getData().getInvitePin().contains("O")).isFalse();
+        assertThat(response.getData().getInvitePin().contains("U")).isFalse();
+        assertThat(response.getData().getInvitePin().contains("0")).isFalse();
+        assertThat(response.getData().getInvitePin().contains("1")).isFalse();
```

Comment:
you can just do `assertThat(response.getData().getInvitePin()).doesNotContain("I","O","U","0","1")`

Reply:
Good point 😄